beacon-chain/db/kv/blocks.go

```diff
@@ -238,41 +278,26 @@ func (k *Store) SaveBlock(ctx context.Context, block *ethpb.BeaconBlock) error {
 func (k *Store) SaveBlocks(ctx context.Context, blocks []*ethpb.BeaconBlock) error {
 	ctx, span := trace.StartSpan(ctx, "BeaconDB.SaveBlocks")
 	defer span.End()
-	encodedValues := make([][]byte, len(blocks))
-	keys := make([][]byte, len(blocks))
-	for i := 0; i < len(blocks); i++ {
-		enc, err := proto.Marshal(blocks[i])
-		if err != nil {
-			return err
-		}
-		key, err := ssz.SigningRoot(blocks[i])
-		if err != nil {
-			return err
-		}
-		encodedValues[i] = enc
-		keys[i] = key[:]
-	}
-	return k.db.Batch(func(tx *bolt.Tx) error {
-		bucket := tx.Bucket(blocksBucket)
-		for i := 0; i < len(blocks); i++ {
-			indicesByBucket := createBlockIndicesFromBlock(blocks[i], tx)
-			if err := updateValueForIndices(indicesByBucket, keys[i], tx); err != nil {
-				return errors.Wrap(err, "could not update DB indices")
-			}
-			k.blockCache.Set(string(keys[i]), blocks[i], time.Hour)
-			if err := bucket.Put(keys[i], encodedValues[i]); err != nil {
-				return err
+	var wg sync.WaitGroup
+	var err error
+	wg.Add(len(blocks))
+	for _, blk := range blocks {
+		go func(w *sync.WaitGroup, b *ethpb.BeaconBlock) {
+			defer w.Done()
+			if err = k.SaveBlock(ctx, b); err != nil {
```

Comment:
This might cause errors to be overridden. Instead of having a single error value, have you considered another way to track errors?

I'm just thinking that if we launched 3 go routines and they executed in this order
1) err = nil 
2) err = "something bad happened"
3) err = nil

the final result is that err == nil, which is a silent failure